src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -36,6 +36,9 @@ internal class HttpInListener : ListenerHandler
         private readonly bool hostingSupportsW3C = false;
         private readonly AspNetCoreInstrumentationOptions options;
 
+        // Create an ActivitySource here to re-create Activity.
+        private ActivitySource activitySource = new ActivitySource("Microsoft.AspNetCore");
```

Comment:
>ActivitySource [](start = 16, length = 14)

do we usually create only one instance of HttpInListener and use it the life time of the app/service?  I am asking to know if we need to explicitly dispose this ActivitySource when it is not needed.

Reply:
Yes only one instance for life of app.

Reply:
good. we should be ok here then. thanks.

Reply:
In the existing implementation, we dispose the DiagnosticListeners when TracerFactory is disposed. Don't think we need it here.